vowpalwabbit/interactions_predict.h

```diff
@@ -71,9 +78,42 @@ inline float INTERACTION_VALUE(float value1, float value2) { return value1 * val
 
 // #define GEN_INTER_LOOP
 
+std::tuple<std::pair<features::const_audit_iterator, features::const_audit_iterator>,
+    std::pair<features::const_audit_iterator,
+        features::const_audit_iterator>> inline generate_quadratic_char_combination(const std::array<features,
```

Comment:
nit: the spacing here is hard to read. Can we go with something like
```
inline generate_quadratic_char_combination(
    const std::array<features, NUM_NAMESPACES>& feature_groups,
    namespace_index ns_idx1,
    namespace_index ns_idx2
)
```
?

Reply:
I'd prefer to just use what the formatter says. I think this is better after the typedef